### PR TITLE
Prune dependency warnings and update readme instructions.

### PR DESCRIPTION
**Pull request by VersusFacit**
## Description & motivation

In going through the docs site setup, I encountered a number of difficulties and concerns regarding dependencies. After much experimentation to figure out exactly what was causing the errors present in the `npm install` log, I am confident that the modifications included in this PR:
1. reduce the number of `npm install` warnings to as few as possible
2. make first-time setup easier for new contributors by noting technical prerequisites 
3. (try to) keep new contributors from breaking the package-lock.json, which I did on my local machine a number of times as I was trying to figure out what, if anything, was wrong with docusaurus.

This does not have a matching issue.

## Pre-release docs
Not related to an unreleased version of dbt. First parent is `current.`

image

### Explanation of Docusaurus Dependencies for Later

Docusaurus packages are all set to complain that the version 2.0.0 is missing. There's one problem: that version doesn't exist. The package is still in alpha releases. Hence, there is no way to satisfy these requirements. It took me quite a while to figure out this was happening. I've included notes in the README to hopefully avoid anyone else in the future attempting this, or worse, ruining their `package.json` such that they are unable to make changes to the website without there being plenty of errors in the logs about how all packages must match the docusaurus core version.

A timely `npm audit fix` was performed this PR as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,14 +12,21 @@ We welcome contributions from community members to this repo:
...
-4. Install the required node packages: `npm install`
-5. Build the website: `npm start`
-6. Before pushing your changes to a branch, check that all links work by using the `make build` script.
+### Running the Docs site locally
+
+We recommend locally rendering changes made to the docs site so you can review your proposed modifications. Our setup instructions use [homebrew](https://brew.sh/):
+
+0. If applicable, install [Xcode CLTs](https://developer.apple.com/download/more/); you'll likely need an AppleID for this.
+1. Install `node`: `brew install node`
+2. Install `yarn`: `brew install yarn`
```
Comment (clrcrl) [inline README.md]: Are both needed? I was under the impression it was one or the other

**Review by clrcrl — APPROVED** [review]
One small comment that I don't know the answer to!
Otherwise LGTM 🎉

File: README.md (hunk 1 above)
Comment (VersusFacit) [reply README.md]: oops! It's only necessary if you attempt to solve the docusaurus issue, which I did. >.>

Removed--updated version being pushed. Good catch!

**Comment by clrcrl** [comment]
Feel free to merge whenever :)

Generally we follow the convention of:
1. Person A opens PR
2. Person B approves
3. Person A merges

(Except for community-contributed PRs, where we have to merge for them)

**Pull request merged**